src/mode/mode_struct.rs

```diff
@@ -1215,10 +1270,43 @@ impl<T: Transformation> Mode<T> {
         } else {
             pepped_up_control_value.calc_distance_from(current_target_value.to_continuous_value())
         };
+
+        // If there is no previous pepped-up control value, set one
+        let previous_pepped_up_control_value = match self.state.previous_pepped_up_control_value {
+            None => pepped_up_control_value,
+            Some(value) => value,
+        };
+        self.state.previous_pepped_up_control_value = Some(pepped_up_control_value);
+
+        let time_expired = match self.state.previous_control_value_time {
+            None => true,
+            Some(time) => time.elapsed().as_millis() > CONTROL_MOVE_TIMEOUT,
+        };
+
+        let in_sync = self.is_in_sync(
+            self.settings.jump_interval.max_val(),
+            time_expired,
+            previous_pepped_up_control_value.to_unit_value(),
+            pepped_up_control_value.to_unit_value(),
+            current_target_value.to_unit_value(),
+            self.state.takeover_in_sync,
+        );
+
+        // Check for controller jumps. For fast movements, the time won't be expired between messages.
+        // For slow movements, the delta between the previous and current values will be under Jump Max.
+        let control_has_jumped = time_expired &&
+            (pepped_up_control_value.to_unit_value().get() -
```

Comment:
Here you can subtract unit values directly again. And it would be good to introduce some intermediate variables to not extract the unit value again and again (you do it also a bit further up).